src/west/manifest.py

```diff
@@ -908,18 +908,20 @@ def from_file(source_file: Optional[PathType] = None,
                 # neither source_file nor topdir: search the filesystem
                 # for the workspace and use its manifest.path.
                 topdir = util.west_topdir()
+                mpath = _mpath(topdir=topdir)
                 kwargs.update({
                     'topdir': topdir,
-                    'source_file': os.path.join(topdir, _mpath(topdir=topdir),
-                                                _WEST_YML)
+                    'source_file': os.path.join(topdir, mpath, _WEST_YML),
+                    'manifest_path': mpath
                 })
             else:
                 # Just source_file: find topdir starting there.
                 # We need source_file in kwargs as that's what gets used below.
                 kwargs.update({
                     'source_file': source_file,
                     'topdir':
-                    util.west_topdir(start=os.path.dirname(source_file))
+                    util.west_topdir(start=os.path.dirname(source_file)),
+                    'manifest_path': os.path.dirname(source_file)
```

Comment:
This assumes the manifest directly contains the source file. That is true for foo/west.yml but it is not true in the general case the API is trying to solve. Why isn't this calling `_mpath()` too?

Reply:
because calling `_mpath` would return the west config value `manifest.path`.

But when calling `Manifest.from_file(source_file=<source_file>)` , then that file can be located anywhere, and in that case the value of `manifest.path` which is returned by `_mpath` might be wrong.

So as soon source file is provided directly, then `_mpath` cannot be used reliable.

Reply:
as example, changing this in `_mapth` would cause the mpath in this comment: https://github.com/zephyrproject-rtos/west/pull/422#discussion_r470450611 to become `mp`, as that is the value of `manifest.path`.

But `another_yaml` is located directly at `topdir`, therefore in this case it is more correct to use `topdir`.

**note** if `source_file` is not provided, as example: `Manifest.from_file()` or `Manifest.from_file(topdir=<topdir>)` then `_mpath` is of course used.